File: src/mb/mizinkobusters/kitpvp/kit/Thunder.java
package mb.mizinkobusters.kitpvp.kit;

import mb.mizinkobusters.kitpvp.utils.KitPvPUtils;
import org.bukkit.Material;
import org.bukkit.attribute.Attribute;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.event.player.PlayerRespawnEvent;
import org.bukkit.inventory.ItemStack;

import java.util.Random;

public class Thunder implements Listener {

	@EventHandler
	public void onKill(PlayerRespawnEvent event) {
		Player player = event.getPlayer();
		if (player.getKiller() == null) {
			return;
		}

		Player killer = player.getKiller();
		if (!KitPvPUtils.isInWorld(killer)) {
			return;
		}
		if (!KitPvPUtils.getKit(killer).equals("Thunder")) {
			return;
		}
		killer.getInventory().addItem(new ItemStack(Material.GOLDEN_APPLE));
		killer.setHealth(killer.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue());
	}

	@EventHandler
	public void onDamage(EntityDamageByEntityEvent event) {
		if (!event.getEntity().getType().equals(EntityType.PLAYER)) {
			return;
		}

		Player damagee = (Player) event.getEntity();
		if (!KitPvPUtils.isInWorld(damagee)) {
			return;
		}

		if (!(event.getDamager().getType().equals(EntityType.PLAYER))) {
			return;
		}

		Player damager = (Player) event.getDamager();
		if (!KitPvPUtils.getKit(damager).equals("Thunder")) {
			return;
		}

		Random r = new Random();
		int i = r.nextInt(6);
		if (i == 0) {
			damagee.getWorld().strikeLightningEffect(damagee.getLocation());
			damagee.damage(event.getDamage() * 1.2, damager);
			damager.sendMessage("§7Skill Trigger: §eThunder");
		}
	}
}

File: src/main/java/mb/mizinkobusters/kitpvp/kit/Thunder.kt
package mb.mizinkobusters.kitpvp.kit

import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.entity.Player
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.inventory.ItemStack

object Thunder : BaseKit {
    override val kitTypeInfo = "§dTarget"
    override val lore = listOf(
        "§f頭: 鉄",
        "§f胴: 金",
        "§f腰: 鉄",
        "§f脚: 金",
        "§f武器: 金の剣[ダメージ増加II]",
        "§f補助: 金のリンゴ×2",
        "§f特殊: 1/7の確率で与ダメージの1.2倍のダメージを与える雷を相手の頭上に落とす"
    )
    override val displayItemStack = ItemStack(Material.DOUBLE_PLANT)
    override val helmet = ItemStack(Material.IRON_HELMET)
    override val chestplate = ItemStack(Material.GOLD_CHESTPLATE)
    override val leggings = ItemStack(Material.IRON_LEGGINGS)
    override val boots = ItemStack(Material.GOLD_BOOTS)
    override val weapons = listOf(
        ItemStack(Material.GOLD_SWORD).apply {
            addEnchantment(Enchantment.DAMAGE_ALL, 2)
        },
        ItemStack(Material.GOLDEN_APPLE, 2)
    )

    override fun onDamageByEntity(event: EntityDamageByEntityEvent) {
        super.onDamageByEntity(event)

        val entity = event.entity
        if (entity !is Player) return

        val damager = event.damager

        if ((0..6).random() == 0) {
            entity.world.strikeLightningEffect(entity.location)
            entity.damage(event.damage * 1.2, damager)
            entity.sendMessage("§7Skill Trigger: §eThunder")
        }
    }
}